packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -112,127 +50,37 @@ window.\$hotReloadHook = function(modules) {
-         if (!errorCallbacks.has(moduleName)) {
-           errorCallbacks.set(moduleName, new Set());
-         }
-         errorCallbacks.get(moduleName).add(onError);
-       }
-       requirejs.undef(moduleName);
-       requirejs([moduleName], function() {
-         if (typeof onError != 'undefined') {
-           errorCallbacks.get(moduleName).delete(onError);
-         }
-         if (typeof callback != 'undefined') {
-           callback();
-         }
-       });
-     },
-     getModuleLibraries: null, // set up by _initializeTools
-   };
-}
-let modulePaths = {};
-let customModulePaths = {};
-window.\$dartLoader.rootDirectories.push(window.location.origin + baseUrl);
-for (let moduleName of Object.getOwnPropertyNames(modulePaths)) {
-  let modulePath = modulePaths[moduleName];
-  if (modulePath != moduleName) {
-    customModulePaths[moduleName] = modulePath;
-  }
-  var src = window.location.origin + '/' + modulePath + '.js';
-  if (window.\$dartLoader.moduleIdToUrl.has(moduleName)) {
-    continue;
-  }
-  \$dartLoader.moduleIdToUrl.set(moduleName, src);
-  \$dartLoader.urlToModuleId.set(src, moduleName);
-}
 // Create the main module loaded below.
 define("main_module.bootstrap", ["$entrypoint", "dart_sdk"], function(app, dart_sdk) {
   dart_sdk.dart.setStartAsyncSynchronously(true);
   dart_sdk._debugger.registerDevtoolsFormatter();
-  \$dartLoader.getModuleLibraries = dart_sdk.dart.getModuleLibraries;
-  if (window.\$dartStackTraceUtility && !window.\$dartStackTraceUtility.ready) {
-    window.\$dartStackTraceUtility.ready = true;
-    let dart = dart_sdk.dart;
-    window.\$dartStackTraceUtility.setSourceMapProvider(
-      function(url) {
-        url = url.replace(baseUrl, '/');
-        var module = window.\$dartLoader.urlToModuleId.get(url);
-        if (!module) {
-          if (url.endsWith('dart_sdk.js')) {
-            module = 'dart_sdk';
-          } else {
-            try {
-              module = '/packages' + url.split('packages')[1].split('.')[0] + '.dart';
-            } catch (e) {
-              return null;
-            }
-          }
-        }
-        return dart.getSourceMap(module);
-      });
-  }
+  dart_sdk._isolate_helper.startRootIsolate(() => {}, []);
   if (typeof document != 'undefined') {
     window.postMessage({ type: "DDC_STATE_CHANGE", state: "start" }, "*");
   }
-  dart_sdk._isolate_helper.startRootIsolate(() => {}, []);
 
   // See the generateMainModule doc comment.
   var child = {};
   child.main = app[Object.keys(app)[0]].main;
-  if (window.\$hotReload == null) {
-    window.\$hotReload = function(cb) {
-      dart_sdk.developer.invokeExtension("ext.flutter.disassemble", "{}").then((_) => {
-        dart_sdk.dart.hotRestart();
-        window.\$dartRunMain();
-        window.requestAnimationFrame(cb);
-      });
-    }
-  }
 
   /* MAIN_EXTENSION_MARKER */
   child.main();
-});
 
-// Require JS configuration.
-require.config({
-  waitSeconds: 0,
+window.\$dartLoader = {};
```

Comment:
Is the dartLoader stuff required? Shouldn't be anymore.

Reply:
We still need this since we're using the stack trace mapper from the SDK, and that code unconditionally refers to these getters

Reply:
Oh yes. Hmmm. We should reconsider that at one point.